app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -100,6 +111,39 @@ class SparkRestClient(sparkConf: SparkConf) {
     }
   }
 
+  private def getLogData(attemptTarget: WebTarget): Option[SparkLogDerivedData] = {
+    val target = attemptTarget.path("logs")
+    logger.info(s"calling REST API at ${target.getUri} to get eventlogs")
+
+    // The logs are stored in a ZIP archive with a single entry.
+    // It should be named as "$logPrefix.$archiveExtension", but
+    // we trust Spark to get it right.
+    resource.managed { getApplicationLogs(target) }.acquireAndGet { zis =>
+      val entry = zis.getNextEntry
+      if (entry == null) {
+        logger.warn(s"failed to resolve log for ${target.getUri}")
+        None
+      } else {
+        val codec = SparkLogClient.compressionCodecForLogName(sparkConf, entry.getName)
+        Some(SparkLogClient.findDerivedData(
```

Comment:
I was thinking about the timeouts and I came to the conclusion that the time to fetch the log does not depend on its size (unless I'm missing something), because `findDerivedData` looks up the first occurrence of `SparkListenerEnvironmentUpdate`, which should be one the first entries in the log.

This makes me wonder, what could be causing the timeouts if not the size? @shkhrgpt any ideas?